components/scream/src/dynamics/homme/atmosphere_dynamics.cpp

```diff
@@ -570,89 +558,50 @@ void HommeDynamics::homme_pre_process (const int dt) {
-        FQ(icol,iq,ilev) = Q(icol,iq,ilev) - FQ(icol,iq,ilev);
-      });
-    }
-    // Remap FT, FM, and Q (or FQ, depending on ftype)
+    // Remap FT, FM, and Q->FQ
     m_p2d_remapper->remap(true);
   }
 
   auto& tl = c.get<TimeLevel>();
-  auto& ff = c.get<ForcingFunctor>();
-
-  const auto& tracers = c.get<Tracers>();
-  const auto& state = c.get<ElementsState>();
-  auto Q    = tracers.Q;
-  auto FQ   = tracers.fq;
-  auto Qdp  = tracers.qdp;
-  auto dp3d = state.m_dp3d;
 
   // Note: np1_qdp and n0_qdp are 'deduced' from tl.nstep, so the
   //       following call may not even change them (i.e., they are
   //       not updated regardless). So if they were already up-to-date,
   //       the following call will do nothing.
   tl.update_tracers_levels(params.qsplit);
 
-  const auto n0 = tl.n0;  // The time level where pd coupling remapped into
-  constexpr int NVL = HOMMEXX_NUM_LEV;
-  const int qsize = params.qsize;
-  const auto n0_qdp = tl.n0_qdp;
-  const auto Q_dyn = m_helper_fields.at("Q_dyn").get_view<Homme::Scalar*****>();
-
   // At this point, FQ contains Qnew (coming from physics).
   // Depending on ftype, we are going to do different things:
   //  ftype=0: FQ = dp*(Qnew-Qold) / dt
-  //  ftype=2: qdp = dp*Qnew, with Qnew stored in FQ
-  switch(ftype) {
-    case ForcingAlg::FORCING_0:
-      // Back out tracers tendency for Qdp
-      Kokkos::parallel_for(Kokkos::RangePolicy<>(0,Q.size()),KOKKOS_LAMBDA(const int idx) {
-        const int ie = idx / (qsize*NP*NP*NVL);
-        const int iq = (idx / (NP*NP*NVL)) % qsize;
-        const int ip = (idx / (NP*NVL)) % NP;
-        const int jp = (idx / NVL) % NP;
-        const int k  =  idx % NVL;
-
-        // fq is currently storing q_new
-        const auto& q_prev = Q(ie,iq,ip,jp,k);
-        const auto& dp     = dp3d(ie,n0,ip,jp,k);
-              auto& fq     = FQ(ie,iq,ip,jp,k);
-
-        fq -= q_prev;
-        fq /= dt;
-        fq *= dp;
-      });
-      Kokkos::fence();
-      break;
-    case ForcingAlg::FORCING_2:
-      if (not fv_phys_active()) {
-        Kokkos::parallel_for(Kokkos::RangePolicy<>(0,Q.size()),KOKKOS_LAMBDA(const int idx) {
-          const int ie = idx / (qsize*NP*NP*NVL);
-          const int iq = (idx / (NP*NP*NVL)) % qsize;
-          const int ip = (idx / (NP*NVL)) % NP;
-          const int jp = (idx / NVL) % NP;
-          const int k  =  idx % NVL;
-
-          // So far, FQ contains the remap of Q_ref_new - Q_ref_old. Add Q_dyn_old to get new state.
-          FQ(ie,iq,ip,jp,k) += Q_dyn(ie,iq,ip,jp,k);
-        });
-      }
-      // Hard adjustment of qdp
-      ff.tracers_forcing(dt,n0,n0_qdp,true,params.moisture);
-      break;
-    default:
-      EKAT_ERROR_MSG ("Error! Unexpected/unsupported forcing algorithm.\n"
-                      "  ftype: " + std::to_string(Homme::etoi(ftype)) + "\n");
+  //  ftype=2: nothing
+  if (ftype == ForcingAlg::FORCING_0) {
```

Comment:
@bartgol this is the tendency computation for ftype=0 for both np4 and pg2 physics.

Reply:
Oh, ftype2 is the hard adjustment, I forgot.